src/templates/base-generated-config.yaml

```diff
@@ -11,18 +11,31 @@ jobs:
         type: string
       test-suite-path:
         type: string
+      bb-version:
+        type: string
     machine:
       image: ubuntu-2004:current
       docker_layer_caching: true
     steps:
       - checkout
+      - run:
+          name: Checkout Test Suite from Specific Version
+          command: |
+            # Clone repository into a temporary directory and switch to the specified tag
+            git clone . temp_repo
+            cd temp_repo
+            git checkout tags/<< parameters.bb-version >>
+            rm -rf << parameters.test-suite-path >>
+            cp -r test/* << parameters.test-suite-path >>
+            cd .. && rm -rf temp_repo
```

Comment:
Okay I fixed that

Reply:
Logic from python script is changed to take only tags and app name as arguments and then fetch test-suite directories from specific tag. Rest of the code is similar to the code before this commit.

Next in template there was added a check to avoid issues with accessing directories that does not exist.